go.mod

```diff
@@ -0,0 +1,23 @@
+module k8s.io/release
+
+go 1.12
```

Comment:
aside: while this doesn't really do much now, we do need go 1.11 to work still I think.

Reply:
1. Do we need to support go:1.11? Note: This is to compile stuff in k/release, and not to compile k8s itself -- for that, the kubecross image is used
1. That was added by go, but yeah, I did the migration with go:1.12
1. It doesn't seem to bother `golang:1.11`, it seems to happily find the deps and generally compile
1. We already have a hard dependency on go:1.12 for the relnotes tool

----

```sh
hhoerl@pfah[±|gomod → origin ✓]:~/src/release $ podman run --rm -ti -v `pwd`:/tmp/foo golang:1.11 bash -c 'cd /tmp/foo && ./hack/test-go.sh'
go: finding github.com/blang/semver v3.5.1+incompatible
  [ ... downloading the internet ... ]
go: downloading golang.org/x/text v0.3.2
=== RUN   TestGetKubeadmConfig
--- PASS: TestGetKubeadmConfig (0.00s)
=== RUN   TestGetKubeadmDependencies
=== RUN   TestGetKubeadmDependencies/minimum_supported_kubernetes
=== RUN   TestGetKubeadmDependencies/latest_stable_minor_kubernetes
--- PASS: TestGetKubeadmDependencies (0.00s)
    --- PASS: TestGetKubeadmDependencies/minimum_supported_kubernetes (0.00s)
    --- PASS: TestGetKubeadmDependencies/latest_stable_minor_kubernetes (0.00s)
PASS
ok      k8s.io/release/build/debs       1.024s
# k8s.io/release/pkg/notes
pkg/notes/notes.go:252:11: undefined: strings.ReplaceAll
pkg/notes/notes.go:253:10: undefined: strings.ReplaceAll
pkg/notes/notes.go:352:14: undefined: strings.ReplaceAll
pkg/notes/notes.go:666:9: undefined: strings.ReplaceAll
note: module requires Go 1.12# k8s.io/release/pkg/notes [k8s.io/release/pkg/notes.test]
pkg/notes/notes.go:252:11: undefined: strings.ReplaceAll
pkg/notes/notes.go:253:10: undefined: strings.ReplaceAll
pkg/notes/notes.go:352:14: undefined: strings.ReplaceAll
pkg/notes/notes.go:666:9: undefined: strings.ReplaceAll
note: module requires Go 1.12?          k8s.io/release/cmd/blocking-testgrid-tests      [no test files]
FAIL    k8s.io/release/pkg/notes [build failed]
hhoerl@pfah[±|gomod → origin ✓]:~/src/release $ 
```

Reply:
> Do we need to support go:1.11? Note: This is to compile stuff in k/release, and not to compile k8s itself -- for that, the kubecross image is used

are all of these tools used exclusively against this repo itself for the forseeable future? I imagine we'll want to rewrite tools used by k/k in go eventually ...

>  That was added by go, but yeah, I did the migration with go:1.12

Right, it also only matters as far as signaling intent. (Same comment for 3)

> We already have a hard dependency on go:1.12 for the relnotes tool

Hmm that should possibly its own module and / or repo anyhow.

Reply:
>> Do we need to support go:1.11? Note: This is to compile stuff in k/release, and not to compile k8s itself -- for that, the kubecross image is used

> are all of these tools used exclusively against this repo itself for the forseeable future? I imagine we'll want to rewrite tools used by k/k in go eventually ...

1. I think for the foreseeable future they will just be used against k/release
1. Soon, when 1.13 is gone, we only have go1.12 anyway 


>> We already have a hard dependency on go:1.12 for the relnotes tool

> Hmm that should possibly its own module and / or repo anyhow.

I think it's generally fine to be in k/release. Shouldn't you advertise for monorepos? 😏

Reply:
> I think it's generally fine to be in k/release. Shouldn't you advertise for monorepos? smirk

monorepos are interesting if they're truly a monorepo and you have the tooling to support them (which ... E.G. github is not so much), a mono repo in the midst of many repos is kinda pointless. The google monorepo doesn't involve downloading the entire state of the world to your machine :wink: 

breaking up modules within a repo is pretty nice, similar to breaking a package into many packages. in particular if I only touch relnotes then I only need to deal with / download / update relnotes's dependencies etc. ... 

kind is currently 3 modules. :upside_down_face: